src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java

```diff
@@ -30,71 +32,90 @@ public final class ThrowableToErrorMapper {
       (throwable, message) -> {
         // if our own exception, shortcut
         if (throwable instanceof JsonApiException jae) {
-          return jae.getCommandResultError(message);
+          return jae.getCommandResultError(message, Response.Status.OK);
         }
-        // Override response error code
+
+        // construct fieldsForMetricsTag, only expose exceptionClass in debugMode
         SmallRyeConfig config = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
         DebugModeConfig debugModeConfig = config.getConfigMapping(DebugModeConfig.class);
         final boolean debugEnabled = debugModeConfig.enabled();
         Map<String, Object> fields =
             debugEnabled ? Map.of("exceptionClass", throwable.getClass().getSimpleName()) : null;
         final Map<String, Object> fieldsForMetricsTag =
             Map.of("exceptionClass", throwable.getClass().getSimpleName());
+
+        // Authentication Failure -> ErrorCode.UNAUTHORIZED_REQUEST
         if (throwable instanceof UnauthorizedException
             || throwable
                 instanceof com.datastax.oss.driver.api.core.servererrors.UnauthorizedException) {
-          return new CommandResult.Error(
-              "UNAUTHENTICATED: Invalid token",
-              fieldsForMetricsTag,
-              fields,
-              Response.Status.UNAUTHORIZED);
+          return ErrorCode.UNAUTHORIZED_REQUEST
+              .toApiException()
+              .getCommandResultError(message, Response.Status.UNAUTHORIZED);
```

Comment:
I am not sure: should this be converted to `JsonApiException`, if it needs to produce non-200 HTTP Response code?

I guess we just need to decide which way to go. I guess there are some benefits in making everything `JsonApiException`, but not necessarily HTTP 200 responses.

Reply:
Yeah, that is why the above method has status added.
I think currently design for Authentication Failure is to return JsonApiException with 401.

Reply:
This may make sense, as it is more of a "hard" failure, and matches what HTTP 401 response is about.